marlowe/src/Language/Marlowe/Analysis/Safety/Types.hs

```diff
@@ -91,7 +109,254 @@ data SafetyError =
+      , "role-name" .= String (T.decodeUtf8 $ fromBuiltin tokenName)
+      , "fatal" .= True
+      ]
+  toJSON (InvalidCurrencySymbol (CurrencySymbol currencySymbol)) =
+    object
+      [ "error" .= ("InvalidCurrencySymbol" :: String)
+      , "detail" .= ("This currency symbol is not the 28-bytes required by the ledger rules." :: String)
+      , "currency-symbol" .= toJSON (EncodeBase16 $ fromBuiltin currencySymbol)
+      , "fatal" .= True
+      ]
+  toJSON (TokenNameTooLong (TokenName tokenName)) =
+    object
+      [ "error" .= ("TokenNameTooLong" :: String)
+      , "detail" .= ("This token name is longer than the 32 bytes allowed by the ledger rules." :: String)
+      , "token-name" .= String (T.decodeUtf8 $ fromBuiltin tokenName)
+      , "fatal" .= True
+      ]
+  toJSON (InvalidToken token) =
+    object
+      [ "error" .= ("InvalidToken" :: String)
+      , "detail" .= ("This token associates a name with the ada currency symbol." :: String)
+      , "token" .= token
+      , "fatal" .= True
+      ]
+  toJSON (NonPositiveBalance accountId token) =
+    object
+      [ "error" .= ("NonPositiveBalance" :: String)
+      , "detail" .= ("In the initial state of the contract, ths account has a non-positive balance of this token." :: String)
+      , "account-id" .= accountId
+      , "token" .= token
+      , "fatal" .= True
+      ]
+  toJSON (DuplicateAccount accountId token) =
+    object
+      [ "error" .= ("DuplicateAccount" :: String)
+      , "detail" .= ("In the initial state of the contract, there are duplicate entries for this account with this token." :: String)
+      , "account-id" .= accountId
+      , "token" .= token
+      , "fatal" .= True
+      ]
+  toJSON (DuplicateChoice choiceId) =
+    object
+      [ "error" .= ("DuplicateChoice" :: String)
+      , "detail" .= ("In the initial state of the contract, there are duplicate entries for this choice." :: String)
+      , "choice-id" .= choiceId
+      , "fatal" .= True
+      ]
+  toJSON (DuplicateBoundValue valueId) =
+    object
+      [ "error" .= ("DuplicateBoundValue" :: String)
+      , "detail" .= ("In the initial state of the contract, there are duplicate entries for this bound value." :: String)
+      , "value-id" .= valueId
+      , "fatal" .= True
+      ]
+  toJSON (MaximumValueMayExceedProtocol natural) =
+    object
+      [ "error" .= ("MaximumValueMayExceedProtocol" :: String)
+      , "detail" .= ("At some point in during its executation, the contract may hold more native tokens than permitted by the ledger rules." :: String)
+      , "bytes" .= natural
+      , "fatal" .= False
+      ]
+  toJSON (TransactionSizeMayExceedProtocol transaction natural) =
+    object
+      [ "error" .= ("TransactionSizeMayExceedProtocol" :: String)
+      , "detail" .= ("This transaction's size may exceed the limit permitted by the ledger rules." :: String)
+      , "transaction" .= transaction
+      , "bytes" .= natural
+      , "fatal" .= False
```

Comment:
Marked as non-fatal because this is an estimate, and the slow analysis would identify if this truly occurs in any transaction.